featurebyte/api/api_object.py

```diff
@@ -544,3 +544,34 @@ def info(self, verbose: bool = False) -> Dict[str, Any]:
             info["class_name"] = self.__class__.__name__
             return InfoDict(info)
         raise RecordRetrievalException(response, "Failed to retrieve object info.")
+
+    def update_description(self, description: str) -> None:
+        """
+        Update description of an object
+
+        Parameters
+        ----------
+        description: str
+            Description of the object
+
+        Raises
+        ------
+        RecordUpdateException
+            When unexpected record update failure
+        """
+        client = Configurations().get_client()
+        response = client.patch(
+            url=f"{self._route}/{self.id}/description", json={"description": description}
+        )
+        if response.status_code == HTTPStatus.OK:
+            object_dict = response.json()
+            self._update_cache(object_dict)  # update object cache
+            type(self).__init__(
+                self,
+                **object_dict,
+                **self._get_init_params_from_object(),
+            )
+        elif response.status_code == HTTPStatus.NOT_FOUND:
+            self.description = description
+        else:
+            raise RecordUpdateException(response=response)
```

Comment:
Would it better to update `def update` method to accept `url` parameter, and then call it by:
```
self.update(
    update_payload={"description": description},
    allow_update_local=False,
    url=f"{self._route}/{self.id}/description",
)
```

Doing this, we could avoid setting description before the object is saved (`allow_update_local=False`) and also reuse update logic (like update cached handling, exception handling etc).

Reply:
We may also need to add a `description` property here to retrieved the description value. Something like
```
@property
def description(self):
        return self.cached_model.description:
```